src/ledger/parse/utils.js

```diff
@@ -58,13 +58,32 @@ function isPartialPayment(tx) {
 function parseDeliveredAmount(tx: Object): Amount | void {
   let deliveredAmount;
 
-  // TODO: Workaround for existing rippled bug where delivered_amount may not be
-  // provided for account_tx
-  if (tx.TransactionType === 'Payment') {
-    if (tx.meta.delivered_amount) {
+  if (tx.TransactionType === 'Payment' &&
+      tx.meta.TransactionResult === 'tesSUCCESS') {
+
+    if (tx.meta.delivered_amount &&
+        tx.meta.delivered_amount === 'unavailable') {
+      deliveredAmount = 'unavailable';
+
+    // parsable delivered_amount
+    } else if (tx.meta.delivered_amount) {
       deliveredAmount = parseAmount(tx.meta.delivered_amount);
+
+    // DeliveredAmount only present on partial payments
+    } else if (tx.meta.DeliveredAmount) {
+      deliveredAmount = parseAmount(tx.meta.DeliveredAmount);
+
+    // no partial payment flag
     } else if (tx.Amount && !isPartialPayment(tx)) {
       deliveredAmount = parseAmount(tx.Amount);
+
+    // partial payment after 4594094 without DeliveredAmount
+    } else if (tx.Amount && tx.ledger_index > 4594094) {
+      deliveredAmount = parseAmount(tx.Amount);
```

Comment:
nope, partial payment flag means it can be successful even if its not the
full amount, but `DeliveredAmount` only shows up if it was less than the
full amount - so after 4594094 a tx with partial payment flag without
DeliveredAmount means it delivered the full amount.

On Mon, Jul 18, 2016 at 12:00 PM, Alan Cohen <notifications@github.com>
wrote:

> In src/ledger/parse/utils.js
> <https://github.com/ripple/ripple-lib/pull/725#discussion_r71207518>:
>
> >      } else if (tx.Amount && !isPartialPayment(tx)) {
> >        deliveredAmount = parseAmount(tx.Amount);
> > +
> > +    // partial payment after 4594094 without DeliveredAmount
> > +    } else if (tx.Amount && tx.ledger_index > 4594094) {
> > +      deliveredAmount = parseAmount(tx.Amount);
>
> If it's a partial payment, doesn't that mean the delivered amount is *not*
> equal to tx.Amount?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ripple/ripple-lib/pull/725/files/9eb0fd3a03ed01c38920225869ec6edb47030f17#r71207518>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AD0McCoScVwpcbDRGIy3dUQ9wLkJMzurks5qW806gaJpZM4JPBrq>
> .
>